sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -188,45 +174,44 @@ private static void WriteMemory(MemoryStream stream, ReadOnlyMemory<byte> memory
         /// ]]></code>
         /// </example>
         /// <remarks> See error response code and error response message for more detail. </remarks>
-        public virtual Response<UploadLogsResult> Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response Upload<T>(string ruleId, string streamName, IEnumerable<T> logs, UploadLogsOptions options = null, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(ruleId, nameof(ruleId));
             Argument.AssertNotNullOrEmpty(streamName, nameof(streamName));
             Argument.AssertNotNullOrEmpty(logs, nameof(logs));
 
             using var scope = ClientDiagnostics.CreateScope("LogsIngestionClient.Upload");
 
-            RequestContext requestContext = GenerateRequestContext(cancellationToken);
             Response response = null;
-            List<UploadLogsError> errors = new List<UploadLogsError>();
+            var exceptions = new List<Exception>();
+
             scope.Start();
-            try
+            // Partition the stream into individual blocks
+            foreach (BatchedLogs<T> batch in Batch(logs, options)) //TODO: check if we should throw early for sync case? as soon as we find exception
             {
-                // Partition the stream into individual blocks
-                foreach (BatchedLogs<T> batch in Batch(logs, options))
+                try
                 {
                     // Because we are uploading in sequence, wait for each batch to upload before starting the next batch
-                    BatchUpload task = CommitBatchListSyncOrAsync(
+                    response = CommitBatchListSyncOrAsync(
                         batch,
                         ruleId,
                         streamName,
                         false,
```

Comment:
nit - for boolean constants like this it's very helpful to pass it via a named parameter like `async: false`